File: test/fixtures/code-map/sample.py
"""Sample Python file for code-map scanner tests."""
import os
from .y import a, b

MAX_BATCH_SIZE = 100
DEBUG_MODE = False

# A simple class
class Foo:
    """Foo class."""

    def __init__(self, name: str) -> None:
        self.name = name

    async def fetch(self, url: str) -> str:
        """Async fetch method."""
        return url

def simple_function(x: int) -> int:
    """A simple module-level function."""
    return x * 2

@route('/x')
def decorated_function(request):
    """A decorated function."""
    return request

File: test/fixtures/code-map/sample.js
// sample.js — fixture for code-map scanner tests
import { foo, bar } from './utils.js';
import defaultExport from 'some-module';
import 'side-effect-module';

const require = createRequire(import.meta.url);
require('legacy-module');

const MAX_RETRIES = 3;
const API_BASE_URL = 'https://example.com';

export class Bar {
  compute(x) {
    return x * 2;
  }
}

export const handler = async () => {
  const result = await fetch(API_BASE_URL);
  return result.json();
};

export function declared() {
  return 'declared function';
}

const helper = function namedHelper() {
  return 42;
};

export default {
  Bar,
  handler,
  declared,
  helper,
};
